### PR TITLE
Add a relative timestamp when tracing.

### PR DESCRIPTION
**Pull request by baldersheim**
make it testable by allowing control of clock.
@havardpe PR

### CODE REVIEW

File: searchlib/src/tests/engine/searchapi/searchapi_test.cpp [hunk 1]
```diff
@@ -235,39 +235,44 @@ void verify(vespalib::stringref expected, const vespalib::Slime & slime) {
...
            "        {"
-           "            tag: 'tag_a'"
+           "            tag: 'tag_a',"
+           "            time: 1"
            "        }"
            "    ],"
            "    creation_time: 7"
            "}",
            t.getSlime());
+#if 0
```
Comment (havardpe) [inline searchlib/src/tests/engine/searchapi/searchapi_test.cpp]: why disable test?

File: searchlib/src/vespa/searchlib/engine/request.cpp [hunk 2]
```diff
@@ -5,8 +5,8 @@
 
 namespace search::engine {
 
-Request::Request(const fastos::TimeStamp &start_time)
-    : _startTime(start_time),
+Request::Request(RelativeTime && relativeTime)
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/engine/request.cpp]: should probably pass by value here as well

**Review by havardpe — COMMENTED** [review]
consider using shared_ptr instead of CloneablePtr, or maybe unique_ptr and moving everywhere is better.

File: searchlib/src/tests/engine/searchapi/searchapi_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/tests/engine/searchapi/searchapi_test.cpp]: Accident, thanks.

File: searchlib/src/vespa/searchlib/engine/request.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/engine/request.cpp]: Fixed

**Comment by baldersheim** [comment]
I tried with unique_ptr, but found 2 places where it could not be moved.
I shall revisit that code agai, and try a bit harder.

**Pull request merged**